### PR TITLE
[feature] Added controller integration for wireguard #150

### LINKED ISSUE #150: [feature] Add controller integration for Wireguard
This module as an optional subapp which depends on openwisp-controller and is able to connect the topology information to the Device in OpenWISP Controller, that allows to show meaningful names in the topology data.

The code is here: https://github.com/openwisp/openwisp-network-topology/tree/master/openwisp_network_topology/integrations/device.

The differences for Wireguard should be contained in the methods mentioned here:
https://github.com/openwisp/openwisp-network-topology/blob/master/openwisp_network_topology/integrations/device/base/models.py#L25-L31

I think the up and down actions for wireguard can be empty. The `auto_create_wireguard` shall look into the DB for devices of the same organization (or any organization if the topology is shared) for devices which have a related vpnclient object with the same ip, eg: 

```
Device.objects.filter(config__vpnclient__ip__ip_address='10.254.0.5').first()
```

We shall update [the docs regarding this integration](https://github.com/openwisp/openwisp-network-topology#integration-with-openwisp-controller-and-openwisp-monitoring) to mention Wireguard.

### PR DESCRIPTION
**Pull request by codesankalp**
Closes #150

### CODE REVIEW

File: openwisp_network_topology/integrations/device/base/models.py [hunk 1]
```diff
@@ -69,21 +90,28 @@ def auto_create_openvpn(cls, node):
...
+    def auto_create_wireguard(cls, node):
+        allowed_ips = node.properties.get('allowedIps')
+        if not allowed_ips:
             return
-        else:
-            logger.info(f'DeviceNode relation created for {node.label} - {device.name}')
-            return device_node
+        Device = load_model('config', 'Device')
+        ip_addresses = [str(IPv4Network(cidr).network_address) for cidr in allowed_ips]
+        device_filter = models.Q(config__vpnclient__ip__ip_address__in=ip_addresses)
```
Comment (codesankalp) [inline openwisp_network_topology/integrations/device/base/models.py:102]: I used allowed Ips in this case and modified query accordingly.

Comment (nemesisdesign) [reply openwisp_network_topology/integrations/device/base/models.py:102]: the ip address is going to be only the ip address, while the allowed ip of wireguard is CIDR notation

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Can you please fix the failing builds on python 3.7?

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
https://github.com/openwisp/openwisp-network-topology/pull/148 has been merged, please rebase on the latest master.

File: README.rst [hunk 2]
```diff
@@ -434,12 +434,12 @@ Integration with OpenWISP Controller and OpenWISP Monitoring
...
 or `OpenWISP Monitoring <https://github.com/openwisp/openwisp-monitoring>`_
-and you use OpenVPN for the management VPN, you can use the integration available in
-``openwisp_network_topology.integrations.device``.
+and you use OpenVPN or Wireguard for the management VPN, you can use the integration
+available in ``openwisp_network_topology.integrations.device``.
 
 This additional and optional module provides the following features:
 
-- whenever the stauts of an OpenVPN link changes:
+- whenever the status of an OpenVPN link changes:
```
Comment (nemesisdesign) [inline README.rst:442]:
```suggestion
- whenever the status of a link changes:
```

**Review by nemesisdesign — APPROVED** [review]
Great progress, it works!

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**